From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Henry Lin <henryl@nvidia.com>
Date: Tue, 11 Nov 2025 07:40:21 +0000
Subject: [PATCH 155/243] NVIDIA: SAUCE: usb: gadget: xudc: Add Tegra264
 virtualization config

Add SOC data for Tegra264 virtualization config.

Bug 5332406

Change-Id: I8f2000e53ed656138aa474efdd58ff02be66cc82
Signed-off-by: Henry Lin <henryl@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3487589
Reviewed-by: EJ Hsu <ejh@nvidia.com>
Reviewed-by: svcacv <svcacv@nvidia.com>
Reviewed-by: Brad Griffis <bgriffis@nvidia.com>
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
(cherry picked from commit 423df67df668f3ef1c4a71e3a4195605e0f967c1)

diff --git a/drivers/usb/gadget/udc/tegra-xudc.c b/drivers/usb/gadget/udc/tegra-xudc.c
index 522a8b89b8184..6dea5b07e1886 100644
--- a/drivers/usb/gadget/udc/tegra-xudc.c
+++ b/drivers/usb/gadget/udc/tegra-xudc.c
@@ -3833,6 +3833,21 @@ static struct tegra_xudc_soc tegra264_xudc_soc_data = {
 	.xudc_device_params_init = tegra264_xudc_device_params_init,
 };
 
+static struct tegra_xudc_soc tegra264_hv_xudc_soc_data = {
+	.num_clks = 0, /* no clock access */
+	.num_phys = 4,
+	.u1_enable = true,
+	.u2_enable = true,
+	.lpm_enable = true,
+	.invalid_seq_num = false,
+	.pls_quirk = false,
+	.port_reset_quirk = false,
+	.has_ipfs = false,
+	.has_pg_support = true,
+	.xudc_device_params_init = tegra264_xudc_device_params_init,
+	.is_hv_guest = true,
+};
+
 static const struct of_device_id tegra_xudc_of_match[] = {
 	{
 		.compatible = "nvidia,tegra210-xudc",
@@ -3858,6 +3873,10 @@ static const struct of_device_id tegra_xudc_of_match[] = {
 		.compatible = "nvidia,tegra264-xudc",
 		.data = &tegra264_xudc_soc_data
 	},
+	{
+		.compatible = "nvidia,tegra264-hv-xudc",
+		.data = &tegra264_hv_xudc_soc_data
+	},
 	{ }
 };
 MODULE_DEVICE_TABLE(of, tegra_xudc_of_match);
-- 
2.43.0

